build_tools/docker/dockerfiles/mmperf.Dockerfile

```diff
@@ -0,0 +1,120 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# An image for running mmperf: https://github.com/mmperf/mmperf.
+#
+# mmperf benchmarks matrix-multiplication workloads on IREE and various backends
+# such as OpenBLAS, MKL, TVM, Halide, CuBLAS, etc.
+#
+# These backends are included either in this image or as a submodule in the
+# mmperf repo. Later versions of Clang, LLVM, Python and Ubuntu are needed
+# to satisfy the dependency requirements of the backends.
+
+# Ubuntu 22.04.
+FROM ubuntu@sha256:817cfe4672284dcbfee885b1a66094fd907630d610cab329114d036716be49ba
+
+######## Basic ########
+RUN apt-get update \
+  && apt-get install -y \
+    git \
+    wget \
+    cmake \
+    curl \
+    ninja-build
+##############
+
+######## Clang/LLVM ########
+RUN apt-get update \
+  && apt-get install -y \
+    llvm-14 \
+    llvm-14-dev \
+    clang-14 \
+    clang-tools-14 \
+    libclang-common-14-dev \
+    libclang-14-dev \
+    libclang1-14 \
+    clang-format-14 \
+    clangd-14 \
+    clang-tidy-14 \
+    lldb-14 \
+    lld-14 \
+    libmlir-14-dev \
+    mlir-14-tools \
+  && ln -s /usr/lib/llvm-14/bin/clang /usr/bin/clang \
+  && ln -s /usr/lib/llvm-14/bin/clang++ /usr/bin/clang++
+##############
+
+######## Python ########
+RUN apt-get update \
+  && apt-get install -y \
+    python3.10 \
+    python3.10-dev \
+  && update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.10 1 \
+  && apt-get install -y \
+    python3-pip \
+    python3-setuptools \
+    python3-distutils \
+    python3-venv \
+    python3.10-venv \
+  && python3 -m pip install --upgrade pip>=21.3 \
+  && python3 -m pip install --upgrade setuptools \
+  && python3 -m pip install --ignore-installed \
+    requests
```

Comment:
`--ignore-installed` here shouldn't be necessary. That's only present in the other docker container because of pyyaml. Also, could you share code with https://github.com/iree-org/iree/blob/a30cc085662d1bfbc522ff831ff9fabf0738ad73/build_tools/docker/context/install_python_deps.sh? It could take the requirements file as an arg (and then you can use input redirection to just make that `echo requirements`)

Reply:
Done